bundles/core/src/main/java/com/adobe/cq/commerce/core/components/utils/SiteNavigation.java

```diff
@@ -14,24 +14,33 @@
 
 package com.adobe.cq.commerce.core.components.utils;
 
+import java.util.Iterator;
+
 import javax.annotation.Nullable;
 
+import org.apache.commons.lang3.ArrayUtils;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.tuple.Pair;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.day.cq.commons.inherit.HierarchyNodeInheritanceValueMap;
 import com.day.cq.commons.inherit.InheritanceValueMap;
+import com.day.cq.commons.jcr.JcrConstants;
+import com.day.cq.wcm.api.NameConstants;
 import com.day.cq.wcm.api.Page;
 import com.day.cq.wcm.api.PageManager;
+import com.day.cq.wcm.api.WCMMode;
 
 public class SiteNavigation {
```

Comment:
Not 100% related to this research but I think we should have this URL generation an OSGI service. So customers can bring their own implementation to generate PDP / PLP URLs based on their needs?

Reply:
Yes it's a good idea, customers will surely want to customise that.